cmd/skopeo/inspect.go

```diff
@@ -86,21 +90,37 @@ func (opts *inspectOptions) run(args []string, stdout io.Writer) (retErr error)
 		return err
 	}
 
-	img, err := parseImage(ctx, opts.image, imageName)
+	sys, err := opts.image.newSystemContext()
+	if err != nil {
+		return err
+	}
+
+	src, err := parseImageSource(ctx, opts.image, imageName)
 	if err != nil {
 		return err
 	}
 
+	closer := io.Closer(src)
```

Comment:
At least, use `imgCloser` or something like that, please; to make it clear what it is for.

Or, maybe (weakly preferred?), use `image.FromUnparsedImage(…, UnparsedInstance(…))` instead of `image.FromSource`. `image.FromSource` more or less exists only to preserve the older design where `types.Image` has a `Close` method (so that callers of `types.ImageReference.NewImage` don’t have to deal with closing an `ImageSource`); like other places that need both an `ImageSource` and `Image`, that design no longer results in simpler code here. (An explicit `UnparsedInstance(src, nil)` would also be a tiny bit more explicit about what is happening WRT multi-arch.)